tests/drop_counters/configurable_drop_counters.py

```diff
@@ -0,0 +1,163 @@
+
+        counters[line[0]] = int(line[1])
+        reasons[line[0]] = []
+        line = output.pop(0)
+
+    # Extract the supported drop reasons from the second part of the output.
+    curr_type = []
+    for line in output:
+        if not line:
+            continue
+        elif line in SUPPORTED_COUNTER_TYPES:
+            curr_type = reasons[line]
+        else:
+            curr_type.append(line.strip())
+
+    return {"counters": counters, "reasons": reasons}
+
+def create_drop_counter(dut, counter_name, counter_type, drop_reasons):
+    """
+    Creates a drop counter on the target device.
+
+    Args:
+        dut (SonicHost): The target device.
+        counter_name (str): The name of the counter to be created.
+        counter_type (str): The type of counter to create.
+        drop_reasons (List[str]): The drop reasons to add to the counter.
+
+    Raises:
+        RunAnsibleModuleFail: If the given counter name is already in use.
+    """
+
+    dut.command(_CREATE_COUNTER.format(counter_name, counter_type, ",".join(drop_reasons)))
+
+def delete_drop_counter(dut, counter_name):
+    """
+    Deletes the drop counter from the target device.
+
+    Args:
+        dut (SonicHost): The target device.
+        counter_name (str): The name of the counter to be created.
+    """
+
+    dut.command(_DELETE_COUNTER.format(counter_name))
+
+def get_drop_counts(dut, counter_type, counter_name, interface):
+    """
+    Gets the count for a given counter on a given interface.
+
+    Note:
+        If the specified type is a SWITCH level counter, then the "interface" field is ignored.
+
+    Args:
+        dut (SonicHost): The target device.
+        counter_type (str): The type of counter being queried.
+        counter_name (str): The name of the counter to query.
+        interface (str): The interface to query.
+
+    Returns:
+        The number of drops on the specified counter, or "None" if the counter type is not
+        supported or the counter is not found.
+    """
+
+    if counter_type not in SUPPORTED_COUNTER_TYPES:
+        return None
+
+    bind_point = interface if "PORT" in counter_type else dut.hostname
+
+    output = dut.command("show dropcounters counts -t {}".format(counter_type))["stdout_lines"]
+    counts = _parse_drop_counts(counter_type, output)
+
+    return int(counts[bind_point.upper()].get(counter_name))
+
+def _parse_drop_counts(counter_type, counts_output):
+    leading_rows = _PARSER_PARAMETERS[counter_type].leading_rows
+    header_rows = _PARSER_PARAMETERS[counter_type].header_rows
+    status_columns = _PARSER_PARAMETERS[counter_type].status_columns
+
+    # Skip any leading rows in the output
+    trimmed_output = counts_output[leading_rows:]
+
+    # Skip over the status columns to get the counter names
+    counter_names = trimmed_output[0].split()[status_columns:]
+
+    # Skip over the header rows to get to the actual counts
+    interface_counts = trimmed_output[header_rows:]
```

Comment:
Just curious, why did you have to trim 3 times instead of just add up 3 overheads and skip once?

Reply:
Part of it is because some of the offsets are for the rows and some of them are for the columns, so they need to be handled in different places.

I kept both of the row offsets because I thought it was more clear since logically we:
1. Skip any leading whitespace
2. Get the names of the counters from the first row of "real" output
3. Skip ahead to the actual counts

We could just fetch the counter names and then delete everything before the counts, but I'm not sure if this improves the performance or readability.